components/plugin-manager/implementation/src/main/kotlin/endpoints/GetInstalledPlugins.kt

```diff
@@ -0,0 +1,120 @@
+
+import org.eclipse.apoapsis.ortserver.components.pluginmanager.PluginDescriptor
+import org.eclipse.apoapsis.ortserver.components.pluginmanager.PluginOption
+import org.eclipse.apoapsis.ortserver.components.pluginmanager.PluginOptionType
+import org.eclipse.apoapsis.ortserver.components.pluginmanager.PluginType
+
+import org.ossreviewtoolkit.advisor.AdviceProviderFactory
+import org.ossreviewtoolkit.analyzer.PackageManagerFactory
+import org.ossreviewtoolkit.plugins.advisors.vulnerablecode.VulnerableCodeFactory
+import org.ossreviewtoolkit.plugins.api.PluginDescriptor as OrtPluginDescriptor
+import org.ossreviewtoolkit.plugins.api.PluginOption as OrtPluginOption
+import org.ossreviewtoolkit.plugins.api.PluginOptionType as OrtPluginOptionType
+import org.ossreviewtoolkit.plugins.packageconfigurationproviders.api.PackageConfigurationProviderFactory
+import org.ossreviewtoolkit.plugins.packagecurationproviders.api.PackageCurationProviderFactory
+import org.ossreviewtoolkit.plugins.packagemanagers.node.npm.NpmFactory
+import org.ossreviewtoolkit.reporter.ReporterFactory
+import org.ossreviewtoolkit.scanner.ScannerWrapperFactory
+
+fun Route.getInstalledPlugins() = get("admin/plugins", {
+    operationId = "GetInstalledPlugins"
+    summary = "Get installed ORT plugins"
+    description = "Get a list with detailed information about all installed ORT plugins."
+    tags = listOf("Plugins")
+
+    response {
+        HttpStatusCode.OK to {
+            description = "Success"
+            body<List<PluginDescriptor>> {
+                mediaTypes = setOf(ContentType.Application.Json)
+                example("List of ORT plugins") {
+                    value = listOf(
+                        VulnerableCodeFactory.descriptor.mapToApi(PluginType.ADVISOR),
+                        NpmFactory.descriptor.mapToApi(PluginType.PACKAGE_MANAGER)
+                    )
+                }
+            }
+        }
+    }
+}) {
+    val advisors = AdviceProviderFactory.ALL.values.map {
+        it.descriptor.mapToApi(PluginType.ADVISOR)
+    }
+    val packageConfigurationProviders = PackageConfigurationProviderFactory.ALL.values.map {
+        it.descriptor.mapToApi(PluginType.PACKAGE_CONFIGURATION_PROVIDER)
+    }
+    val packageCurationProviders = PackageCurationProviderFactory.ALL.values.map {
+        it.descriptor.mapToApi(PluginType.PACKAGE_CURATION_PROVIDER)
+    }
+    val packageManagers = PackageManagerFactory.ALL.values.map {
+        it.descriptor.mapToApi(PluginType.PACKAGE_MANAGER)
+    }
+    val reporters = ReporterFactory.ALL.values.map {
+        it.descriptor.mapToApi(PluginType.REPORTER)
+    }
+    val scanners = ScannerWrapperFactory.ALL.values.map {
+        it.descriptor.mapToApi(PluginType.SCANNER)
+    }
+
+    val allPlugins = advisors +
+            packageConfigurationProviders +
+            packageCurationProviders +
+            packageManagers +
+            reporters +
+            scanners
```

Comment:
The code to get ORT plugin information will be moved to a dedicated class in a follow-up PR as part of adding more functionality, I just left it here to keep the first version simple.